modular_skyrat/master_files/code/modules/pain_shock/status_indicator.dm

```diff
@@ -0,0 +1,187 @@
+		value = myclient?.prefs?.read_preference(/datum/preference/toggle/enable_status_indicators)
+	var/atom/movable/screen/plane_master/status/status = locate() in myclient?.screen
+	if(!status)
+		return
+	if(value && status)
+		status.Show()
+	else
+		status.Hide()
+
+/mob/living
+	var/list/status_indicators = null // Will become a list as needed. Contains our status indicator objects. Note, they are actually added to overlays, this just keeps track of what exists.
+
+/// Returns true if the mob is weakened. Also known as floored.
+/mob/living/proc/is_weakened()
+	if(HAS_TRAIT(src, TRAIT_FLOORED) || has_status_effect(/datum/status_effect/incapacitating/knockdown && !HAS_TRAIT_FROM(src, TRAIT_FLOORED, BUCKLED_TRAIT)))
+		return TRUE
+
+/// Returns true if the mob is stunned.
+/mob/living/proc/is_stunned()
+	if(HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || HAS_TRAIT(src, TRAIT_CRITICAL_CONDITION) || HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, CHOKEHOLD_TRAIT)  || HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(STAT_TRAIT)))
+		return TRUE
+
+/// Returns true if the mob is paralyzed - for can't fight back purposes.
+/mob/living/proc/is_paralyzed()
+	if(HAS_TRAIT_FROM(src, TRAIT_FLOORED, CHOKEHOLD_TRAIT) || HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || HAS_TRAIT(src, TRAIT_CRITICAL_CONDITION)  || HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, STAMINA))
+		return TRUE
+
+/// Returns true if the mob is unconcious for any reason.
+/mob/living/proc/is_unconcious()
+	if(HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+		return TRUE
+
+/// Returns true if the mob has confusion.
+/mob/living/proc/is_confused()
+	if(has_status_effect(/datum/status_effect/confusion))
+		return TRUE
+
+/mob/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_MOB_CLIENT_LOGIN, .proc/apply_status_indicator_pref)
+
+/mob/living/carbon/Initialize(mapload)
+	. = ..()
+	// The Basics
+	RegisterSignal(src, COMSIG_CARBON_HEALTH_UPDATE, .proc/status_indicator_evaluate)
+	RegisterSignal(src, COMSIG_LIVING_DEATH, .proc/cut_indicators_overlays)
+	// Catching weird cases
+	RegisterSignal(src, COMSIG_CARBON_HELP_ACT, .proc/status_indicator_evaluate)
+	RegisterSignal(src, COMSIG_LIVING_SET_BODY_POSITION, .proc/status_indicator_evaluate)
+	// When things actually happen
+	RegisterSignal(src, COMSIG_LIVING_STATUS_STUN, .proc/status_indicator_evaluate)
+	RegisterSignal(src, COMSIG_LIVING_STATUS_KNOCKDOWN, .proc/status_indicator_evaluate)
+	RegisterSignal(src, COMSIG_LIVING_STATUS_PARALYZE, .proc/status_indicator_evaluate)
+	RegisterSignal(src, COMSIG_LIVING_STATUS_IMMOBILIZE, .proc/status_indicator_evaluate)
+	RegisterSignal(src, COMSIG_LIVING_STATUS_UNCONSCIOUS, .proc/status_indicator_evaluate)
+	RegisterSignal(src, COMSIG_LIVING_STATUS_SLEEP, .proc/status_indicator_evaluate)
```

Comment:
pretty sure there is a way to cut down on this.

Reply:
Ah, man, these are all the signals used where I would want to evaluate the mob. I think it's better than over-riding procs.

Reply:
I've cut a few! let me know!